novawallet/Common/DataProvider/NPoolsLocalSubscriptionFactory.swift

```diff
@@ -268,7 +268,9 @@ extension NPoolsLocalSubscriptionFactory: NPoolsLocalSubscriptionFactoryProtocol
             endTimestamp.map { "\($0)" } ?? "nil"
         ].joined(separator: "-")
 
-        let identifier = ("poolReward" + api.url.absoluteString) + address + timeIdentifier
+        let identifier = ("poolReward" + api.map(\.url.absoluteString).joined(with: .dash))
```

Comment:
Let's hash the identifier. Also i think we need to sort apis by url since we have a set as parameter and it can give different order for the same set